system_metrics_collector/test/system_metrics_collector/test_periodic_measurement_node.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <iostream>
+#include <string>
+#include <memory>
+#include <mutex>
+
+#include "../../src/system_metrics_collector/collector.hpp"
+#include "../../src/system_metrics_collector/periodic_measurement_node.hpp"
+#include "../../src/moving_average_statistics/types.hpp"
+
+/**
+ * Simple extension to test basic functionality
+ */
+class TestPeriodicMeasurementNode : public PeriodicMeasurementNode
+{
+public:
+  TestPeriodicMeasurementNode(
+    const std::string name,
+    const std::chrono::milliseconds & measurement_period,
+    const std::string & publishing_topic)
+  : PeriodicMeasurementNode(name, measurement_period, publishing_topic)
+  {}
+  virtual ~TestPeriodicMeasurementNode() = default;
+
+  // made public to manually test
+  void periodicMeasurement() override
+  {
+    sum += 1;
+    acceptData(static_cast<double>(sum.load()));
+  }
+
+private:
+  std::atomic<int> sum{0};
+  std::condition_variable_any cv;
```

Comment:
```suggestion
  std::condition_variable_any cv{};
```

Reply:
This is actually leftover testing cruft, removing.